docs/user-guide/topics/ab_testing.md

```diff
@@ -113,37 +114,40 @@ Track identifiers are mapped to a static set of endpoints. One approach is to ma
 
 === "Node.js"
     ```javascript
-    const trackToRoute = {
-        "backend":   "http://backend.default.svc.cluster.local:8091",
-        "backend-candidate-1": "http://backend-candidate-1.default.svc.cluster.local:8091",
-    }
+    const versionNumberToRoute = [
+        "http://backend.default.svc.cluster.local:8091",
+        "http://backend-candidate-1.default.svc.cluster.local:8091",
+    ]
     ```
 
 === "Go"
     ```go
-    trackToRoute = map[string]string{
-        "backend":             "http://backend.default.svc.cluster.local:8091",
-        "backend-candidate-1": "http://backend-candidate-1.default.svc.cluster.local:8091",
-    }
+	versionNumberToRoute = []string{
+		"http://backend.default.svc.cluster.local:8091",
+		"http://backend-candidate-1.default.svc.cluster.local:8091",
+	}
     ```
 
 ### Using *Lookup()*
 
-Given a user session identifier, *Lookup()* returns a track identifier that can be used to route requests. In code sample below, the user session identifier is assumed to be passed in the `X-User` header of user requests. The track identifier is used as an index to the `trackToRoute` map defined above. A default is used if the call to *Lookup()* fails for any reason.
+Given a user session identifier, *Lookup()* returns a version number that can be used to route requests. In code sample below, the user session identifier is assumed to be passed in the `X-User` header of user requests. The version number is used as an index to the `versionNumberToRoute` map defined above. A default is used if the call to *Lookup()* fails for any reason.
 
 === "Node.js"
     ```javascript
     var application = new messages.Application();
     application.setName('default/backend');
     application.setUser(req.header('X-User'));
-    client.lookup(application, function(err, session) {
-        if (err || (session.getTrack() == '')) {
+    client.lookup(application, function(err, versionRecommendation) {
+        if (err) {
             // use default route (see above)
-            console.warn("error or null")
+            console.warn("error calling Lookup()")
         } else {
             // use route determined by recommended track
-            console.info('lookup suggested track %s', session.getTrack())
-            route = trackToRoute[session.getTrack()];
+            console.info('lookup suggested track %d', versionRecommendation.getVersionnumber())
+            versionNumber = versionRecommendation.getVersionnumber()
+            if (versionNumber != NaN && 0 <= versionNumber && versionNumber < versionNumberToRoute.length) {
```

Comment:
```suggestion
            if (versionNumber != NaN && versionNumber <= 0 && versionNumber < versionNumberToRoute.length) {
```

I think the previous pattern is strange looking. I also see this occur in at least one other place.